src/tests/test_certstor_system.cpp

```diff
@@ -131,6 +158,34 @@ Test::Result find_certs_by_subject_dn_and_key_id(Botan::Certificate_Store& certs
    return result;
    }
 
+Test::Result find_all_certs_by_subject_dn(Botan::Certificate_Store& certstore)
+   {
+   Test::Result result("System Certificate Store - Find all Certificates by subject DN");
+
+   try
+      {
+      auto dn = get_dn();
+
+      result.start_timer();
+      auto certs = certstore.find_all_certs(dn, std::vector<uint8_t>());
+      result.end_timer();
+
+      if(result.confirm("result not empty", !certs.empty()) &&
+            result.test_eq("exactly one certificate", certs.size(), 1))
+         {
+         auto cns = certs.front()->subject_dn().get_attribute("CN");
+         result.test_is_eq("exactly one CN", cns.size(), size_t(1));
+         result.test_eq("CN", cns.front(), "DST Root CA X3");
```

Comment:
This fails because Ubuntu certstore has two certs with this DN. Maybe add the key id also? Or just verify >= 1 instead of == 1

Reply:
This was supposed to be a regression test for the deduplication ...
I was assuming that two certificates with the same DN are the same. If this is not the case, the test doesn't make that much sense I guess (a test with the key id already exists)